mlrun/runtimes/function.py

```diff
@@ -1203,10 +1100,12 @@ def compile_function_config(function: RemoteRuntime, client_version: str = None)
     # Add secret configurations to function's pod spec, if secret sources were added.
     # Needs to be here, since it adds env params, which are handled in the next lines.
     # This only needs to run if we're running within k8s context. If running in Docker, for example, skip.
+    inside_k8s = False
     if get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster():
```

Comment:
You have `get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster()` to tell you whether you're inside k8s or not, why you need another variable for that ?